src/Modules/UngradedModules.jl

```diff
@@ -0,0 +1,3653 @@
+  @show_name(io, F)
+  @show_special(io, F)
+
+  print(io, "Free module of rank $(F.n) over ")
+  print(IOContext(io, :compact =>true), F.R)
+end
+
+@doc Markdown.doc"""
+    rank(F::FreeMod)
+    ngens(F::FreeMod)
+    dim(F::FreeMod)
+
+Return the rank of `F`.
+"""
+dim(F::FreeMod) = F.n
+rank(F::FreeMod) = F.n
+ngens(F::FreeMod) = dim(F)
+
+@doc Markdown.doc"""
+    ==(F::FreeMod, G::FreeMod)
+
+Equality check between `F` and `G`. `F` and `G` are equal iff their 
+base rings, ranks and (basis) element names are equal.
+"""
+function Base.:(==)(F::FreeMod, G::FreeMod)
+  # two free modules are equal if the rank and the ring are
+  # TODO it this enough or e.g. stored morphisms also be considered?
+  return F.R == G.R && rank(F) == rank(G) && F.S == G.S
+end
+
+@doc Markdown.doc"""
+    iszero(F::FreeMod)
+
+Check if `F` is the zero module, that is if `F` has rank 0.
+"""
+function iszero(F::FreeMod)
+  return rank(F) == 0
+end
+
+@doc Markdown.doc"""
+    FreeModElem{T}
+
+The type of free module elements. A free module element
+is determined by a sparse row (`SRow`) which contains the coords 
+wrt to the standard basis and the parent free module.
+
+# Example
+```
+julia> R, (x,y) = PolynomialRing(QQ, ["x", "y"])
+(Multivariate Polynomial Ring in x, y over Rational Field, fmpq_mpoly[x, y])
+
+julia> F = FreeMod(R,2)
+Free module of rank 2 over Multivariate Polynomial Ring in x, y over Rational Field
+
+julia> v = FreeModElem(sparse_row(R, [(1,x),(2,y)]),F)
+(x)*e[1] + (y)*e[2]
+
+julia> v == x*F[1]+y*F[2]
+true
+
+```
+"""
+struct FreeModElem{T}
+  coords::SRow{T} # also usable via coeffs()
+  parent::FreeMod{T}
+end
+
+function in(v::FreeModElem, M::ModuleFP)
+  return parent(v) === M
+end
+
+@doc Markdown.doc"""
+    coords(v::FreeModElem)
+
+Return the entries (wrt to the standard basis) of `v` as a sparse row.
+"""
+function coords(v::FreeModElem)
+  return v.coords
+end
+
+@doc Markdown.doc"""
+    coeffs(v::FreeModElem)
+
+Return the entries (wrt to the standard basis) of `v` as a sparse row.
+"""
+function coeffs(v::FreeModElem)
+  return coords(v)
+end
+
+@doc Markdown.doc"""
+    repres(v::FreeModElem)
+
+Return just `v`. This function exists for compatiblity (with subquotient elements) reasons.
+"""
+function repres(v::FreeModElem)
+  return v
+end
+
+function getindex(v::FreeModElem, i::Int)
+  if isempty(v.coords)
+    return zero(base_ring(v.parent))
+  end
+  return v.coords[i]
+end
+
+elem_type(::Type{FreeMod{T}}) where {T} = FreeModElem{T}
+parent_type(::Type{FreeModElem{T}}) where {T} = FreeMod{T}
+elem_type(::FreeMod{T}) where {T} = FreeModElem{T}
+parent_type(::FreeModElem{T}) where {T} = FreeMod{T}
+
+function show(io::IO, e::FreeModElem)
+  if length(e.coords) == 0
+    print(io, 0)
+    return
+  end
+  i = 1
+  while i <= length(e.coords)
+    print(io, "(", e.coords.values[i], ")*", e.parent.S[e.coords.pos[i]])
+    if i < length(e.coords)
+      print(io, " + ")
+    end
+    i += 1
+  end
```

Comment:
Better:
```suggestion
  for i in 1:length(e.coords)
    print(io, "(", e.coords.values[i], ")*", e.parent.S[e.coords.pos[i]])
    if i < length(e.coords)
      print(io, " + ")
    end
  end
```

Could also go for a one-liner:
```suggestion
  print(io, join(["($val)*$(e.parent.S[pos])" for (pos, val) in e.coords], " + "))
```